### ISSUE TITLE
Prometheus requirements for the machine's CPU and memory

### ISSUE THREAD

**Issue by lwx294821** [prometheus-operator/prometheus-operator#2803]
How much memory and cpu are set by deploying prometheus in k8s?

**Comment by Ghostbaby** [comment]
resources:
    limits:
      cpu: 5000m
      memory: 50Gi
    requests:
      cpu: 3000m
      memory: 400Mi

**Comment by lwx294821** [comment]
@Ghostbaby thanks. The official has instructions on how to set the size?

**Comment by metalmatze** [comment]
I don't think the Prometheus Operator itself sets any requests or limits itself:
https://github.com/coreos/prometheus-operator/blob/04d7a3991fc53dffd8a81c580cd4758cf7fbacb3/pkg/prometheus/statefulset.go#L718-L723

However, in kube-prometheus (which uses the Prometheus Operator) we set some requests:
https://github.com/coreos/kube-prometheus/blob/8405360a467a34fca34735d92c763ae38bfe5917/manifests/prometheus-prometheus.yaml#L19-L21

**Comment by lwx294821** [comment]
thanks @metalmatze

**Comment by MrD2** [comment]
I did some tests and this is where i arrived with the stable/prometheus-operator standard deployments

RAM:: 256 (base) + Nodes * 40 [MB]
CPU:: 128 (base) + Nodes * 7 [mCPU]
Disk:: 15 GB for 2 weeks (needs refinement)

I'm still looking for the values on the DISK capacity usage per number of numMetrics/pods/timesample
Also, on the CPU and memory i didnt specifically relate to the numMetrics. The high value on CPU actually depends on the required capacity to do Data packing.

This memory works good for packing seen between 2 ~ 4 hours window.

Would like to get some pointers if you have something similar so that we could compare values.
I tried this for a 1:100 nodes cluster so some values are strapulated (mainly for the high number of nodes where i would expect that resources stabilize in a log way)
